### PR TITLE
Infrastructure command on bastion of Automate HA cluster for node delete

### PR DESCRIPTION
**Pull request by Sunanda-Boorla**
### :nut_and_bolt: Description: What code changed, and why?
As a user of Automate HA, when I run chef-automate infrastructure and chef-automate infrastructure node-delete command on bastion system of Automate HA cluster, I should see output relevant to Automate HA infrastructure

### :athletic_shoe: How to Build and Test the Change
1. git checkout this branch
2. rebuild components/automate-cli and upload it to you origin
3. Install automate-cli on bastion
4. Add nodes using load test suite.
6. Run `chef-automate infrastructure node-delete <node-id>`

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
image

image

https://progresssoftware-my.sharepoint.com/:v:/g/personal/sboorla_progress_com/EdckQQBzYG1JkOtsf6qNp1oBXXJL0dztWa7PyEpcvJamAw?e=gzoFdo

https://progresssoftware-my.sharepoint.com/:v:/g/personal/sboorla_progress_com/EZln7QaOTvZEtQfgbPvGozEBqXmFWxKsXNvtYRc3KKKIVA?e=Pplfba

https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/ESdkFe-54vFPo-13qc-i1CQBvmUmQQfLdQmypLemy7J2oA?e=YKwNAu

### CODE REVIEW

File: components/automate-cli/cmd/chef-automate/infrastructure_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		testName string
+		cmd      *cobra.Command
+		args     []string
+		wantErr  bool
+	}{
+		{"Test node delete", nodeDelCmd, []string{"uuid of node"}, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+			if err := runDeleteNodeCmd(tt.cmd, tt.args); (err != nil) != tt.wantErr {
```
Comment (jayvikramsharma1) [inline components/automate-cli/cmd/chef-automate/infrastructure_test.go:34]: condition is unnecessarily complex and the if condition also doesn't look correct, please simplify the conditions

File: components/automate-cli/cmd/chef-automate/infrastructure_test.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+				t.Errorf("runDeleteNodeCmd() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+		t.Run(tt.testName, func(t *testing.T) {
+			if err := preInfrastructureCmd(tt.cmd, tt.args); (err != nil) != tt.wantErr {
```
Comment (jayvikramsharma1) [inline components/automate-cli/cmd/chef-automate/infrastructure_test.go:39]: condition is unnecessarily complex and the condition also doesn't look correct, please simplify the conditions

**Review by jayvikramsharma1 — CHANGES_REQUESTED** [review]

File: components/automate-cli/cmd/chef-automate/infrastructure_test.go (hunk 1 above)
Comment (Sunanda-Boorla) [reply components/automate-cli/cmd/chef-automate/infrastructure_test.go]: Done

File: components/automate-cli/cmd/chef-automate/infrastructure_test.go (hunk 2 above)
Comment (Sunanda-Boorla) [reply components/automate-cli/cmd/chef-automate/infrastructure_test.go]: Done

**Review by vivekshankar1 — APPROVED** [review]

**Review by rishabhjhs — APPROVED** [review]

**Review by punitmundra — APPROVED** [review]

**Review by prasad927 — APPROVED** [review]

**Review by ArvinthC3000 — APPROVED** [review]

**Review by vviveksharma — APPROVED** [review]

**Pull request merged**